PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/actions/CreateFunctionAction.java

```diff
@@ -0,0 +1,214 @@
+        try {
+            Azure azure = AuthMethodManager.getInstance().getAzureClient(eventHubNamespace.id().split("/")[2]);
+            EventHubNamespaceAuthorizationRule eventHubNamespaceAuthorizationRule = azure.eventHubNamespaces().
+                    authorizationRules().getByName(eventHubNamespace.resourceGroupName(), eventHubNamespace.name(),
+                    "RootManageSharedAccessKey");
+            return eventHubNamespaceAuthorizationRule.getKeys().primaryConnectionString();
+        } catch (IOException e) {
+            return null;
+        }
+    }
+    private static final String DEFAULT_LOCAL_SETTINGS_JSON = "{\"IsEncrypted\":false,\"Values\":{\"AzureWebJobsStorage\":\"\",\"FUNCTIONS_WORKER_RUNTIME\":\"java\"}}";
+
+    public void saveLocalSetting(Project project, String key, String value) {
+        Gson gson = new GsonBuilder().setPrettyPrinting().disableHtmlEscaping().create();
+        ;
+        JsonObject localSettings = null;
+        final Path localSettingPath = Paths.get(project.getBasePath(), "local.settings.json");
+        synchronized (this) {
+            try (FileInputStream fis = new FileInputStream(localSettingPath.toFile());
+                 InputStreamReader isr = new InputStreamReader(fis)) {
+                localSettings = gson.fromJson(isr, JsonObject.class);
+            } catch (IOException | JsonParseException e) {
+                localSettings = gson.fromJson(DEFAULT_LOCAL_SETTINGS_JSON, JsonObject.class);
+            }
+        }
+        JsonObject appSettings = localSettings.getAsJsonObject("Values");
+        appSettings.addProperty(key, value);
+        try (Writer writer = new FileWriter(localSettingPath.toFile())) {
+            gson.toJson(localSettings, writer);
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+    }
+
+    @NotNull
+    @Override
+    protected PsiElement[] create(@NotNull String s, PsiDirectory psiDirectory) throws Exception {
+        return new PsiElement[0];
+    }
+
+    private static boolean doCheckPackageExists(PsiDirectory directory) {
+        PsiPackage pkg = JavaDirectoryService.getInstance().getPackage(directory);
+        if (pkg == null) {
+            return false;
+        }
+
+        String name = pkg.getQualifiedName();
+        return StringUtil.isEmpty(name) || PsiNameHelper.getInstance(directory.getProject()).isQualifiedName(name);
+    }
+
+
+    @Override
+    protected boolean isAvailable(final DataContext dataContext) {
+        final Project project = CommonDataKeys.PROJECT.getData(dataContext);
+        final IdeView view = LangDataKeys.IDE_VIEW.getData(dataContext);
+        ProjectFileIndex projectFileIndex = ProjectRootManager.getInstance(project).getFileIndex();
+        for (PsiDirectory dir : view.getDirectories()) {
+            if (projectFileIndex.isUnderSourceRootOfType(dir.getVirtualFile(), JavaModuleSourceRootTypes.SOURCES) && doCheckPackageExists(dir)) {
```

Comment:
Can this works when user want to create Function in a new project? or in a  new package(has been created yet)

Reply:
This prevents it, it must be right clicked at a given package(under a source folder)

Reply:
It has the same logic as create new class in idea, so  a new project without source folder will not be able to create class. for creating new package, just replace the hinted package name with the one you wanted, the create wizard will create the package as you typed.